### PR TITLE
Add @tisyn/validate package for boundary IR validation

### PR DESCRIPTION
**Pull request by taras**
## Summary

- New `@tisyn/validate` package with `validateGrammar()` (Level 1), `validateIr()` (Level 1+2 staged), and `assertValidIr()` as the public API
- Hand-written recursive walker replicates exact `classifyNode` + kernel `validate` accept/reject behavior — TypeBox schemas are JSON Schema export artifacts only, not the validation engine
- 133 tests including a parity suite that runs shared fixtures against both the new package and current kernel/IR validation to lock behavioral equivalence

## Test plan

- [ ] `pnpm --filter @tisyn/validate test` — 133 tests (65 unit + 68 parity)
- [ ] `pnpm build` — full monorepo build passes
- [ ] Review parity.test.ts fixtures cover edge cases (extra fields, unknown tisyn values, nested malformed nodes, structural ops with/without Quote data)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/kernel/src/validate.ts [hunk 1]
```diff
@@ -1,176 +1,16 @@
...
-    case "array":
-      return Array.isArray(fields["items"]) ? fields["items"] : [];
-    case "concat":
-      return Array.isArray(fields["parts"]) ? fields["parts"] : [];
-    case "throw":
-      return fields["message"] !== undefined ? [fields["message"]] : [];
-    default:
-      return [];
-  }
+export function validate(expr: TisynExpr): void {
```
Comment (taras) [inline packages/kernel/src/validate.ts]: why are we keeping this? all it does it call assertValidIr
